app/api/api_v1/endpoints/data_sources.py

```diff
@@ -37,6 +37,7 @@ def read_all_data_sources(
 def read_data_source_by_id(
     data_source_id: int,
     db: Session = Depends(deps.get_db),
+    current_user: models.User = Depends(deps.get_current_user),
```

Comment:
We shouldn't limit this endpoint to authenticated users and should remove this part.

Reply:
Yup, got it.